src/checksum/arm64/crc32-arm64.c

```diff
@@ -40,10 +40,13 @@
 
 #define CRC32CX(crc, value) \
     __asm__("crc32cx %w[c], %w[c], %x[v]" : [ c ] "+r"(*&crc) : [ v ] "r"(+value))
+#pragma GCC diagnostic push
```

Comment:
Looking for some feedback with this change. When building with strict on aarch64 we trigger `-Wunused-macro` errors. I wasn't sure about the best approach with fixing this, we could either:
* Delete the unused macros (though might still be valuable to keep around?).
* Add the CPP pragmas (as seen in this commit)
* Manually append `-Wno-unused-macro` when compiling this specific file (I take this approach with on PPC, see https://github.com/wiredtiger/wiredtiger/blob/651387381e3007c6a12d3a78468fb459e3e02dfd/build_cmake/configs/ppc64le/linux/config.cmake#L27). Less invasive, but only fixes the issue in CMake (issue also occurs with autconf).

Reply:
I agree to go with the current code changes.

Reply:
Do we plan to kill AutoTools support though? In that case, maybe it's better to just set `-Wno-unused-macros` in CMake for consistency since we're also doing it for that other file.

Otherwise, I agree with the pragma approach.

Reply:
Good point, we do plan to eventually remove Autotool's though probably not in the immediate future. That will eventually happen in the long term. The intention is to keep them both working for the time being (as to not suddenly drop support for those that currently depend on it).

Reply:
Sounds good. Pragmas it is. 👍